subprojects/docs/src/docs/userguide/javaPlugin.adoc

```diff
@@ -169,8 +169,11 @@ The Java plugin adds a number of <<managing_dependency_configurations,dependency
 [[tab:configurations]]
 ==== Dependency configurations
 
-`compile`::
-_Compile time dependencies._
+[.line-through]#`compile`#(Deprecated)::
+_Compile time dependencies._ Superseded by `implementation`.
```

Comment:
⭕️ We might want to add a mention of `api`, provided by the java-library plugin here.

Reply:
+1 We definitely should mention that here plus the use case. [Here's](https://docs.gradle.org/nightly/userguide/dependency_management_for_java_projects.html#sec:configurations_java_tutorial) how we approach the description in the chapter "Dependency Management for Java Projects".

Reply:
I think it would also be immensely helpful for users to see an actual scenario that _uses_ a specific configuration by example (e.g. JUnit assigned to `testImplementation`). It may seem too obvious for us but we know it's not immediately clear to beginners.

Reply:
Thanks. I've already added a NOTE section to guide users to `java-library` plugin page. I'm not putting them here because this chapter is `javaPlugin.adoc`.

Reply:
Afaict, java-library actually only introduces the api configuration, implementation is added by the java plugin.